### DISCUSSION TITLE
Why the official MinIO Java driver?

### DISCUSSION THREAD

**Discussion by mechite** [minio/minio#17002] [Q&A]
Heya! I'd like to ask what the advantages are to using the official MinIO Java driver versus using the S3 module from the Amazon AWS SDK.

I have already written my infrastructure around the official MinIO driver which seems to essentially be a "worse" version of the S3 driver where you have this stinky overly employed polymorphism everywhere with argument builders, etc; but while this doesn't really affect usability, my main gripe is the inability to have multiple endpoints! Should I be using the S3 driver, or is there really some performance advantage or something for this driver?

Only being able to define one connection endpoint is horrible, though I'm making the bold assumption the client automatically discovers subsequent nodes and will load balance and have reliable failover built in. This is likely the case, but wouldn't it still be better to have this functionality so that a container won't simply crash if the single endpoint provided happened to be offline? Ephemeral uses of MinIO are actually probably quite a common use case, so being able to define multiple endpoints would lead to better sleep.

I asked this question in the official Slack channel but nobody seemed to respond.

![image](https://user-images.githubusercontent.com/50027352/230798243-cc5570fc-6c15-430e-b391-96ccd1ee7c6e.png)

**Comment by klauspost** [comment]
Please consider your use of language when opening issues. Using this language does not promote constructive dialogue.

Closing this issue. You are welcome to open a new with more appropriate/constructive language.
